File: sound_analytics_platform/backend/app/services/label_matching.py
from __future__ import annotations

_EQUIVALENT_LABELS = frozenset({("dog_bark", "dog"), ("dog", "dog_bark")})


def labels_match(predicted: str | None, ground_truth: str | None) -> bool:
    if not predicted or not ground_truth:
        return False
    if predicted == ground_truth:
        return True
    return (predicted, ground_truth) in _EQUIVALENT_LABELS

File: sound_analytics_platform/frontend/src/lib/labelMatching.ts
const EQUIVALENT_LABELS = new Set(["dog_bark|dog", "dog|dog_bark"]);

export function labelsMatch(predicted: string | null | undefined, groundTruth: string | null | undefined): boolean {
  if (!predicted || !groundTruth) return false;
  if (predicted === groundTruth) return true;
  return EQUIVALENT_LABELS.has(`${predicted}|${groundTruth}`);
}

export function hasGroundTruth(row: {
  has_ground_truth?: boolean;
  ground_truth_label?: string | null;
}): boolean {
  return Boolean(row.has_ground_truth && row.ground_truth_label);
}
